props/saber_BC_buttons.h

```diff
@@ -463,7 +463,23 @@ class SaberBCButtons : public PROP_INHERIT_PREFIX PropBase {
       } else {
         thrust_begin_millis_ = millis();
       }
-  }
+
+    // Scroll Presets timer check - avoid beep/wav overlap
+    if (scroll_presets_timer_trigger_ && DelayTimer(scroll_presets_timer_trigger_, last_timer_time_, beep_seq_duration_ * 1000)) {
+        if (scroll_presets_) {
+            SaberBase::DoEffect(EFFECT_NEWFONT, 0);
+        }
+        scroll_presets_timer_trigger_ = false; // Reset the timer flag
+    }
+  }  // Loop()
+
+    bool DelayTimer(bool& trigger_flag, uint32_t& last_timer_time, float duration) {
```

Comment:
It might be better to make this into a class so you don't have to pass in a bunch of arguments, kind of like:

class DelayTimer {
   public:
     bool run();
     void trigger();
  private:
     bool triggered_;
     uint32_t trigger_time_;
     uint32_t duration_;
};

Reply:
ok should be better now?

Reply:
Yes, it looks better, but the DelayTimer() function can be removed now, right?